### PR TITLE
docs: add --env type

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Docs

**Did you add tests for your changes?**
NA
**If relevant, did you update the documentation?**
yes
**Summary**
added type `string` of `--env` flag

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: packages/webpack-cli/README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Options
   -c, --config string      Provide path to a webpack configuration file
   -m, --merge string       Merge a configuration file using webpack-merge
   --progress               Print compilation progress during build
-  --env                    Environment passed to the configuration when it is a function
+  --env string             Environment passed to the configuration when it is a function
```
Comment (anshumanv) [inline packages/webpack-cli/README.md:42]: We should maybe also mention that only values are `production`, 'development', 'none'

Comment (snitin315) [reply packages/webpack-cli/README.md:42]: I think we can add that info in help with `webpack --env --help`. Like we have for `--mode`

Comment (anshumanv) [reply packages/webpack-cli/README.md:42]: Yeah we can do that, and in case the env is invalid we can also show the `env help`

**Comment by anshumanv** [comment]
@snitin315 are you gonna make the changes here?

File: packages/webpack-cli/README.md (hunk 1 above)
Comment (snitin315) [reply packages/webpack-cli/README.md:42]: I see we can pass `webpack --env invalid` and it doesn't throw a warning or something else to let the user know it's invalid. We should implement this. showing `env help` seems good to me.

**Comment by snitin315** [comment]
> @snitin315 are you gonna make the changes here?

No, in future PR.

File: packages/webpack-cli/README.md (hunk 1 above)
Comment (anshumanv) [reply packages/webpack-cli/README.md:42]: Sure

Comment (snitin315) [reply packages/webpack-cli/README.md:42]: @anshumanv maybe you want to pick this up? I'm caught up with some other issues and PRs :smile:

Comment (anshumanv) [reply packages/webpack-cli/README.md:42]: Cool np, focus on core 👍 
I'll PR by tomm

Comment (anshumanv) [reply packages/webpack-cli/README.md:42]: Ah my bad I confused env with mode, env can accept any string so we're good here 👍

**Pull request merged**